ou_dedetai/gui_app.py

```diff
@@ -579,9 +579,17 @@ def set_appimage_symlink(self):
         utils.set_appimage_symlink(self)
         self.update_latest_appimage_button()
 
-    def update_to_latest_appimage(self, evt=None):
+    def start_appimage_update(self):
         self.status("Updating to latest AppImage…")
-        self.start_thread(self.set_appimage_symlink)
+        self.gui.latest_appimage_button.state(['disabled'])
+        evt = '<<AppImageUpdateDone>>'
+        self.root.bind(evt, self.update_latest_appimage_button)
+        self.start_thread(self.update_to_latest_appimage, evt=evt)
+
+    def update_to_latest_appimage(self, evt=None):
+        utils.update_to_latest_recommended_appimage(self)
+        self.root.event_generate(evt)
```

Comment:
correct me if I'm wrong but don't we only need one function for updating the appimage? Do we still need set_appimage_symlink?

For a function spawning a new thread a pattern I've been using is declaring the function you want to spawn inside the function itself. For cases when you won't want to run the function outside of a separate thread. In python that has the added benefit of capturing the function's arguments. Seen below:

```py
def my_function_that_spawns_a_thread(self, arg_1):
  def _run():
    # do work
    self.function(arg_1)
    pass
  self.start_thread(_run)
```